src/compiler/crystal/command.cr

```diff
@@ -462,6 +462,13 @@ class Crystal::Command
         filenames << stdin_filename
       end
 
+      if single_file
+        opts.before_each do |arg|
```

Comment:
Does it distinguish flags that have a separate value?

E.g. would it understand `--exclude-warnings foo.cr` `bar.cr`?

Maybe worth a test.